src/test/java/cz/muni/fi/pa165/PokemonDaoTest.java

```diff
@@ -0,0 +1,298 @@
+import org.springframework.test.context.testng.AbstractTestNGSpringContextTests;
+import org.springframework.test.context.transaction.TransactionalTestExecutionListener;
+import org.springframework.transaction.annotation.Transactional;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import javax.inject.Inject;
+
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.List;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.*;
+
+/**
+ *  @author Oliver Roch
+ */
+
+@Transactional
+@TestExecutionListeners(TransactionalTestExecutionListener.class)
+@ContextConfiguration(classes = KodemonApplicationContext.class)
+public class PokemonDaoTest extends AbstractTestNGSpringContextTests {
+    @Inject
+    private PokemonDao pokemonDao;
+
+    @Inject
+    private TrainerDao trainerDao;
+
+    private Trainer trainer;
+
+    @BeforeMethod
+    void prepare() {
+        assertThat(pokemonDao, is(notNullValue(PokemonDao.class)));
+        pokemonDao.deleteAll();
+        assertThat("pokemonDao.count() != 0L", pokemonDao.count(), is(0L));
+        assertThat(trainerDao, is(notNullValue(TrainerDao.class)));
+        trainerDao.deleteAll();
+        assertThat("trainerDao.count() != 0L", trainerDao.count(), is(0L));
+        prepareTrainer();
+        assertThat("trainer == null", trainer, is(notNullValue()));
+        trainerDao.save(trainer);
+    }
+
+
+    //Save tests
+
+    @Test(expectedExceptions = {DataAccessException.class})
+    void testSaveNullPokemon() {
+        Pokemon pokemon = null;
+        assertThat(pokemon, is(nullValue(Pokemon.class)));
+        pokemonDao.save(pokemon);
+    }
+
+    @Test
+    void testSaveCorrectInstance() {
+        Pokemon pokemon = new Pokemon();
+
+        pokemon.setTrainer(trainer);
+        pokemon.setName("Eevee");
+        pokemon.setLevel(1);
+        pokemon.setNickname("Killer");
+        pokemon.setType(PokemonType.NORMAL);
+
+        pokemonDao.save(pokemon);
+        assertThat("pokemon.getId() == null", pokemon.getId(), is(notNullValue(Long.class)));
+        Pokemon testPokemon = pokemonDao.findOne(pokemon.getId());
+        assertThat(testPokemon, is(equalTo(pokemon)));
+    }
+
+    @Test
+    void testSaveMultipleCorrectInstances() {
+        int numberOfPokemons = 5;
+        List<Pokemon> pokemons = randomPokemons(numberOfPokemons);
+        assertThat(pokemons.size(), is(equalTo(numberOfPokemons)));
+        pokemonDao.save(pokemons);
+        assertThat(pokemonDao.count(), is(equalTo((long) pokemons.size())));
+        assertThat(pokemonDao.findAll(), is(equalTo(pokemons)));
+    }
+
+    @Test(expectedExceptions = {NullPointerException.class})
+    void testSavePokemonWithNullName() {
+        Pokemon pokemon = new Pokemon();
+
+        pokemon.setTrainer(trainer);
+        pokemon.setLevel(1);
+        pokemon.setNickname("Killer");
```

Comment:
To iste, co som pisal matejovi: bolo by lepsie, ak v testoch, ktore testuju funkciu s nejakym `null` fieldom, tak by som explicitne pisal `setField(null)`, pretoze ak to iekto bude citat, tak bez toho, aby sa pozrel do pokemona a videl,z e tam je default value null, tak nema odkial vediet, ze tam je null.